icu4c/source/common/uloc_tag.cpp

```diff
@@ -2093,11 +2093,13 @@ ultag_parse(const char* tag, int32_t tagLen, int32_t* parsedLen, UErrorCode* sta
             if (tagLen < newTagLength) {
                 uprv_free(tagBuf);
                 tagBuf = (char*)uprv_malloc(newTagLength + 1);
+                t->buf = tagBuf; // Change t->buf after the free and before
+                                 // return to avoid the second double free in
+                                 // the destructor.
```

Comment:
Do you by _the destructor_ mean `ultag_close()`? If so, I think it'd be clearer if you wrote that instead. (If not, what do you mean then?)